x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googlevertexai/response/GoogleVertexAiChatCompletionResponseEntity.java

```diff
@@ -0,0 +1,214 @@
+     * For a request like
+     *
+     * <pre>
+     *     <code>
+     *{
+     *   "contents": [
+     *     {
+     *       "role": "user",
+     *       "parts": [
+     *         {
+     *           "text": "Hello!"
+     *         }
+     *       ]
+     *     }
+     *   ],
+     *   "generationConfig": {
+     *     "responseModalities": [
+     *       "TEXT"
+     *     ],
+     *     "temperature": 1,
+     *     "maxOutputTokens": 8192,
+     *     "topP": 0.95
+     *   }
+     * }
+     *     </code>
+     * </pre>
+     *
+     * The response would look like:
+     *
+     * <pre>
+     *     <code>
+     *[{
+     *   "candidates": [
+     *     {
+     *       "content": {
+     *         "role": "model",
+     *         "parts": [
+     *           {
+     *             "text": "Hello there! How"
+     *           }
+     *         ]
+     *       }
+     *     }
+     *   ],
+     *   "usageMetadata": {
+     *     "trafficType": "ON_DEMAND"
+     *   },
+     *   "modelVersion": "gemini-2.0-flash-001",
+     *   "createTime": "2025-04-29T16:55:36.576032Z",
+     *   "responseId": "iAQRaKCUI_D7ld8Pq-aaaaa"
+     * }
+     * ,
+     * {
+     *   "candidates": [
+     *     {
+     *       "content": {
+     *         "role": "model",
+     *         "parts": [
+     *           {
+     *             "text": " can I help you today?\n"
+     *           }
+     *         ]
+     *       },
+     *       "finishReason": "STOP"
+     *     }
+     *   ],
+     *   "usageMetadata": {
+     *     "promptTokenCount": 2,
+     *     "candidatesTokenCount": 11,
+     *     "totalTokenCount": 13,
+     *     "trafficType": "ON_DEMAND",
+     *     "promptTokensDetails": [
+     *       {
+     *         "modality": "TEXT",
+     *         "tokenCount": 2
+     *       }
+     *     ],
+     *     "candidatesTokensDetails": [
+     *       {
+     *         "modality": "TEXT",
+     *         "tokenCount": 11
+     *       }
+     *     ]
+     *   },
+     *   "modelVersion": "gemini-2.0-flash-001",
+     *   "createTime": "2025-04-29T16:55:36.576032Z",
+     *   "responseId": "iAQRaKCUI_D7ld8Pq-aaaaa"
+     * }
+     * ]
+     *     </code>
+     * </pre>
+     */
+    public static ChatCompletionResults fromResponse(Request request, HttpResult response) throws IOException {
+        try (
+            XContentParser parser = XContentFactory.xContent(XContentType.JSON)
+                .createParser(XContentParserConfiguration.EMPTY, response.body())
+        ) {
+            ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser);
+
+            StringBuilder fullText = new StringBuilder();
+
+            while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                Chunk chunk = Chunk.PARSER.apply(parser, null);
+                chunk.extractText().ifPresent(fullText::append);
```

Comment:
This class would be used for non-streaming scenarios. If we get multiple entries in the array could those be for separate input values in the originating request?

Like:

```
{"input": ["text 1", "text 2"]}
```

Would we get 2 items in the array from the upstream server? If so, I don't think we want to combine the text as we'd want to return a list of 2 items below.